code/HabitTracker/app/src/main/java/com/example/habittracker/activities/fragments/HabitInputFragment.java

```diff
@@ -144,7 +144,7 @@ public void onClick(View view) {
                         if (bundle!= null) {
                             oldTitle = bundle.getString("old_habit_title");
                         }
-                        Habit habit = new Habit(title, reason, datePicker.getSetDate(), weekDays, isPublic, SharedInfo.getInstance().getCurrentUser());
+                        Habit habit = new Habit(title, reason, datePicker.getSetDate(), weekDays,0, isPublic, SharedInfo.getInstance().getCurrentUser());
```

Comment:
I am surprised your code works with this line here and that your ordering doesn't get messed up/reset when a habit is edited. Have you tested this? Because this fragment right here is called for both editing and adding habits.

(I'm modifying the editing and it won't even happen in a fragment, but still. Surprised it works as it is now....)

Might I ask, for when I'm solving conflicts, how do I get the current progress of a habit. Would that be stored as an attribute of the object? (Just asking to check if you're populating progress attribute when you get the habit from the database)

Reply:
Because you may be changing the date of the habit started, the progress would be useless to remember anyway because you would have to recalculate it anyway, I will add the update on it here. Should I add the order into the Habit class to make sure?

Reply:
ohh. So to be clear. You are saying that, ANY time a habit is modified, the progress should be set to 0?

Reply:
I think the way you have implemented ordering right now works okay. So I don't think it's necessary to keep order there too.

Reply:
I have changed it to keep the order it had before it was editing now, check newest commit.